src/Simplex/Chat/Store/Direct.hs

```diff
@@ -424,6 +425,19 @@ updatePCCIncognito db User {userId} conn customUserProfileId = do
     (customUserProfileId, updatedAt, userId, pccConnId conn)
   pure (conn :: PendingContactConnection) {customUserProfileId, updatedAt}
 
+updatePendingContactConnectionUserId :: DB.Connection -> UserId -> UserId -> PendingContactConnection -> IO PendingContactConnection
+updatePendingContactConnectionUserId db userId oldUserId conn = do
+  updatedAt <- getCurrentTime
+  DB.execute
+    db
+    [sql|
+      UPDATE connections
+      SET user_id = ?, updated_at = ?
+      WHERE user_id = ? AND connection_id = ?
+    |]
+    (userId, updatedAt, oldUserId, pccConnId conn)
+  pure (conn :: PendingContactConnection) {updatedAt}
```

Comment:
should also unset incognito (and we should also call profile deletion in api in Chat, similar to unsetting connection incognito api - `deletePCCIncognitoProfile`), and not forget to update in record (otherwise it may or break ui's brain)

```suggestion
      SET user_id = ?, custom_user_profile_id = NULL, updated_at = ?
      WHERE user_id = ? AND connection_id = ?
    |]
    (userId, updatedAt, oldUserId, pccConnId conn)
  pure (conn :: PendingContactConnection) {userId = newUserId, customUserProfileId = Nothing, updatedAt}
```